pkg/sql/schemachanger/scbuild/scbuild_test: TestBuildIsMemoryMonitored failed

pkg/sql/schemachanger/scbuild/scbuild_test.TestBuildIsMemoryMonitored [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa8df84e-38dd-4588-a1bf-f3d6c9865b40?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXIvc2NidWlsZDpzY2J1aWxkX3Rlc3Q=) on master @ [ebf92d22a7e4afe2423e681d49c615104b5f0098](https://github.com/cockroachdb/cockroach/commits/ebf92d22a7e4afe2423e681d49c615104b5f0098):


```
=== RUN   TestBuildIsMemoryMonitored
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBuildIsMemoryMonitored3190076657
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:190: 
        pkg/sql/schemachanger/scbuild/scbuild_test_test/pkg/sql/schemachanger/scbuild/builder_test.go:321: (TestBuildIsMemoryMonitored)
        	NOTICE: .ClusterSettings() called via implicit interface ApplicationLayerInterface;
        HINT: consider using .ApplicationLayer().ClusterSettings() instead.
    conditional_wrap.go:194: TIP: consider replacing the test server initialization from:
            ts, ... := serverutils.StartServer(t, ...)
            defer ts.Stopper().Stop(...)
        to:
            srv, ... := serverutils.StartServer(t, ...)
            defer srv.Stopper().Stop(...)
            ts := srv.ApplicationLayer()
        
        See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    conditional_wrap.go:190: 
        pkg/sql/schemachanger/scbuild/scbuild_test_test/pkg/sql/schemachanger/scbuild/builder_test.go:325: (TestBuildIsMemoryMonitored)
        	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
        HINT: consider using .StorageLayer().NodeID() instead.
    builder_test.go:329: 
        	Error Trace:	pkg/sql/schemachanger/scbuild/scbuild_test_test/pkg/sql/schemachanger/scbuild/builder_test.go:329
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/sctestutils/sctestutils.go:88
        	            				pkg/sql/schemachanger/scbuild/scbuild_test_test/pkg/sql/schemachanger/scbuild/builder_test.go:325
        	Error:      	Error "aborted during Replica.Send: context canceled" does not contain "test-sc-build-mon: memory budget exceeded:"
        	Test:       	TestBuildIsMemoryMonitored
        	Messages:   	got a memory usage of: 0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestBuildIsMemoryMonitored3190076657
--- FAIL: TestBuildIsMemoryMonitored (75.74s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBuildIsMemoryMonitored.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36739

--- a/pkg/sql/schemachanger/scbuild/builder_test.go
+++ b/pkg/sql/schemachanger/scbuild/builder_test.go
@@ -292,7 +292,7 @@ func TestBuildIsMemoryMonitored(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderDeadlock(t, "takes too long")
+	skip.UnderDuress(t, "takes too long; creates thousands of tables")
 
 	ctx := context.Background()
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{

